spec/models/ability_spec.rb

```diff
@@ -89,11 +89,103 @@
+          let(:conference_with_open_registration_and_limit_not_excedeed) { create(:conference, registration_limit: '1', registration_period: registration_period) }
+
+          it{ should be_able_to(:new, Registration.new(conference: conference_with_open_registration_and_limit_not_excedeed)) }
+          it{ should be_able_to(:create, Registration.new(conference: conference_with_open_registration_and_limit_not_excedeed)) }
+        end
+
+        context 'when user has not registered and registrations are open with registration_limit_exceeded' do
+          let(:registration_period) { create(:registration_period) }
+          let(:conference_with_open_registration_and_limit_excedeed) { create(:conference, registration_limit: '1', registration_period: registration_period) }
+          let(:registration_for_user_with_open_registration_and_limit_excedeed) { create(:registration, conference: conference_with_open_registration_and_limit_excedeed, user: user2) }
+
+          it{ should_not be_able_to(:new, Registration.new(conference_id: registration_for_user_with_open_registration_and_limit_excedeed.conference_id)) }
+          it{ should_not be_able_to(:create, Registration.new(conference_id: registration_for_user_with_open_registration_and_limit_excedeed.conference_id)) }
+        end
+      end
+
+      context 'when user is a speaker with event confirmed' do
+        context 'when the registration period is not set' do
+          let(:conference_with_speaker_confirmed) { create(:conference) }
+          let(:event_with_speaker_confirmed) { create(:event, state: 'confirmed', program: conference_with_speaker_confirmed.program) }
+
+          before :each do
+            event_with_speaker_confirmed.speakers << user
+          end
+
+          it{ should_not be_able_to(:new, Registration.new(conference: conference_with_speaker_confirmed)) }
+          it{ should_not be_able_to(:create, Registration.new(conference: conference_with_speaker_confirmed)) }
+        end
+
+        context 'And the registration period is set' do
+          let(:registration_period) { create(:registration_period) }
+          let(:conference_with_speaker_confirmed) { create(:conference, registration_period: registration_period) }
+
+          context 'with registration open' do
+            let(:event_with_speaker_confirmed) { create(:event, state: 'confirmed', program: conference_with_speaker_confirmed.program) }
+
+            before :each do
+              event_with_speaker_confirmed.speakers << user
+            end
+
+            it{ should be_able_to(:new, Registration.new(conference: conference_with_speaker_confirmed)) }
+            it{ should be_able_to(:create, Registration.new(conference: conference_with_speaker_confirmed)) }
+
+            context 'and registration_limit_exceeded' do
+              let(:conference_with_open_registration_and_limit_excedeed) { create(:conference, registration_limit: '1', registration_period: registration_period) }
```

Comment:
Based on the previous comments, instead of this let statement, perhaps you can just change the registration_limit in the existing conference?